Build on windows

g4s8 opened the pull request:
#1041 - disabled footprint test on windows

0crat commented:
Job [#1178](https://github.com/zerocracy/farm/issues/1178) is now in scope, role is `REV`

<!-- https://www.0crat.com/footprint/C3NDPUA8L/1700836189, version: 0.22.6, hash: ${buildNumber} -->

0crat commented:
This pull request [#1178](https://github.com/zerocracy/farm/issues/1178) is assigned to @krzyk[/z](https://www.0crat.com/u/krzyk), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/1700839276); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

<!-- https://www.0crat.com/footprint/C3NDPUA8L/1700840307, version: 0.22.6, hash: ${buildNumber} -->

krzyk commented:
@g4s8 actually FootprintTest wasn't failing in windows, in the #1041 there are 2 major issues:
1. Tests failing because of locale that uses `,` as decimal point
2. Test failing because of S3ItemTest (setting modification time for file in windows, I think the issue here is it can't be set when the file is open)

So I would say excluding the test is a temporary solution (todo for actual fix?).

g4s8 commented:
@krzyk yes, I actually doesn't have windows machine right now, so I'm trying to pass AppVeyor build  in PR and add a todo then

krzyk commented on pom.xml:
```diff
@@ -971,5 +971,28 @@ SOFTWARE.
+              <artifactId>maven-surefire-plugin</artifactId>
+              <configuration combine.children="append">
+                <excludes>
+                  <exclude>com.zerocracy.pm.FootprintTest</exclude>
```
@g4s8 this test is not failing the build on Windows (at least I haven't seen any such case)

krzyk commented on src/test/java/com/zerocracy/farm/S3ItemTest.java:
```diff
@@ -41,6 +41,11 @@
  * @version $Id$
  * @since 0.1
  * @checkstyle JavadocMethodCheck (500 lines)
+ * @todo #1041:30min This test is failing on Windows system,
```
@g4s8 not this one :), S3ItemTest is failing because of the problem with setting modification time of a file - this error appears on all windows versions/locales - appveyor was failing because of that.

Decimal separator is a problem for:

*    com.zerocracy.tk.project.TkArtifactTest#rendersHomePage
*   com.zerocracy.tk.project.TkFootprintTest#rendersListOfClaims
*    com.zerocracy.pm.cost.EquityTest#addValue
*    com.zerocracy.tk.TkAppTest#rendersHomePageViaHttp
*    com.zerocracy.tk.profile.TkIdentifyTest#rendersHomePage

g4s8 replied:
@krzyk but looks like all tests were passed on AppVeyor now: https://ci.appveyor.com/project/yegor256/farm/build/152 I disabled only `S3ItemTest`

krzyk replied:
@g4s8 yes, on appveyor only S3ItemTest was the problem. But proshin-roman has a locale with different decimal separator so it will fail for him (but with one less test failing).

I would make a todo to fix reliance on default decimal point for now.

g4s8 replied:
@krzyk got it, please look at new changes https://github.com/zerocracy/farm/pull/1178/commits/40dd8f47d62a7e2d1ff8dc78e539ef7cff806a51

krzyk commented:
@g4s8 unfortunately now branch coverage ratio is too low and build fails in appveyor

0crat commented:
@krzyk[/z](https://www.0crat.com/u/krzyk) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

<!-- https://www.0crat.com/footprint/C3NDPUA8L/1750942270, version: 0.23.1, hash: ${buildNumber} -->

krzyk commented:
@g4s8 build still fails on appveyor with branch coverage error. I would probably test the profile  it out locally with different OS in the activation and see if the override works

g4s8 commented:
@krzyk yes, profile activation works fine, but I can't override coverage limits (posted a question on SO https://stackoverflow.com/q/50946702/1723695), will try to fix it soon.

krzyk commented:
@g4s8 I think the problem might be because:
1. You have `combine.children="override"` the correct one is `combine.self="override"`
2. jacoco config is in two profiles, I don't know what the order of those will be, so which one will override which. So my proposition is:
a) to move the configuration of jacoco into pluginManagement (without any profile), with by default disabled execution (skip=true)
b) in qulice profile enable the check goal
c) in windows profile enable the check goal, and also override the rules

Maybe that will work.

g4s8 commented:
@krzyk thanks, I've moved jacoco config to `pluginManagement` section. Also I fixed jacoco exclusions: they were pointed to test-cases instead of affecting classes, now I replaced test names with target class names. Also I removed `limits` from build-config, so there is no problem with `combine.children="override"` now.

krzyk commented:
@rultor merge

rultor commented:
> @rultor merge

@krzyk Thanks for your request. @g4s8 Please confirm this.

g4s8 commented:
@rultor merge

rultor commented:
> @rultor merge

@g4s8 OK, I'll try to merge now. You can check the progress of the merge [here](http://www.rultor.com/t/15073-399789148)

rultor commented:
> @rultor merge

@g4s8 Done! FYI, the full log is [here](http://www.rultor.com/t/15073-399789148) (took me 16min)

0crat commented:
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @krzyk[/z](https://www.0crat.com/u/krzyk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

<!-- https://www.0crat.com/footprint/C3NDPUA8L/1752237736, version: 0.23.1, hash: ${buildNumber} -->

0crat commented:
The job [#1178](https://github.com/zerocracy/farm/issues/1178) is now out of scope

<!-- https://www.0crat.com/footprint/C3NDPUA8L/1752237746, version: 0.23.1, hash: ${buildNumber} -->

0crat commented:
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

<!-- https://www.0crat.com/footprint/C3NDPUA8L/1752237779, version: 0.23.1, hash: ${buildNumber} -->

ypshenychka commented:
@0crat quality good

0crat commented:
Order was finished, quality is "good": +20 point(s) just awarded to @krzyk[/z](https://www.0crat.com/u/krzyk)

<!-- https://www.0crat.com/footprint/C3NDPUA8L/1760723971, version: 0.23.1, hash: ${buildNumber} -->

0crat commented:
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

<!-- https://www.0crat.com/footprint/C3NDPUA8L/1760725047, version: 0.23.1, hash: ${buildNumber} -->
